### ISSUE TITLE
Exploration share link not working

### ISSUE THREAD

**Issue by ghislaineguerin** [mathesar-foundation/mathesar#3452]
## Description
When creating a share link for explorations and then navigating to the page, I'm getting a "No page found" error.

## Expected behavior
The shared exploration should load.

## To Reproduce
- Go to an exploration and create a share link
- Navigate to the link

**Comment by seancolsen** [comment]
I'm reproducing this. 

![image](https://github.com/mathesar-foundation/mathesar/assets/42411/76282a1b-d817-49a3-a8f7-a6448184670f)

I get the same behavior when viewing the share link as authenticated and anonymous.

**Comment by seancolsen** [comment]
@pavish heads up that I took a quick crack at this today just in case you ended up being short on time. At the very least, reviewing this PR will help you see where the problem is — within `SharedExplorationRoute.svelte` the `database` variable was `undefined`. I made some changes to the child components to allow a missing database, which I _think_ is okay, but I'd like you to take a close look since you know more than me about what functionality to expect from the database share page.
